Lombiq.BaseTheme.Samples/Assets/Styles/abstracts/_mixins.scss

```diff
@@ -0,0 +1,28 @@
+// -----------------------------------------------------------------------------
+// Application-wide Sass mixins.
+// -----------------------------------------------------------------------------
+
+// You can reference "Assets/Styles/abstracts/_variables.scss" here once and then you never have to again. Other .scss
+// files can import this one to have everything. Make sure to import "variables" before teh Base Theme's "mixins",
+// because that imports some default Sass variable declarations and you want them to override your custom values.
+@import "variables";
+@import "../Lombiq.BaseTheme/Assets/Styles/abstracts/mixins";
+
+// In this case we create a mixin that crops a media field into a nice wide banner. It's applied both for sites and blog
+// posts in other files.
+@mixin media-field-banner {
+    // The "mixins" file in the Lombiq.BaseTheme has several utility mixins, including many "gutter" functions that help
+    // with applying uniform spacing between controls on your web site. Unless you have a specific reason not to, we
+    // suggest always using these to set gap (for grid and flexbox), margin and padding.
+    @include margin-bottom-gutter($multiplier: 1);
```

Comment:
VS says undeclared variable (and mixin) here.

Reply:
This isn't a variable, it is a parameter label just like in C#, so that means the Sass support of VS is busted. (Wouldn't be the first of its kind, their Vue support is so bad it made me want to commit sudoku.)
The mixin is correctly declared in "../Lombiq.BaseTheme/Assets/Styles/abstracts/mixins" which is imported above. If the mixin wasn't declared StyleLint would complain.

Reply:
I know VS is busted and the code is correct as far as the Sass compiler is concerned. Nevertheless, please look into what may cause it (may be something in the syntax somewhere that the Sass compiler excepts but the VS IDE doesn't) because this will be a continuous annoyance for VS users. It would be great if we could get rid of it (or at least open a bug report).

Reply:
The busted is the reason. 😄 I can make a bug report, but I have never did it for VS. Where do you do that?

Reply:
I mean, probably it's some idiosyncrasy of VS that can be worked around with some minimal changes, and the IDE is not fundamentally broken. (Like slashes in the path or something.)

See: https://docs.microsoft.com/en-us/visualstudio/ide/how-to-report-a-problem-with-visual-studio?view=vs-2022

Reply:
Okay, I've found two different problems:

First, sass resolves the `@import "../Lombiq.BaseTheme/Assets/Styles/abstracts/mixins";` above, but VS does not. Looks like it supports path lookup where `.` is sass' working directory (so the project directory where the _package.json_ file is), while VS only supports paths relative to the current _scss_ file's location. This is quite cumbersome, but I can replace it with `@import "../../../../Lombiq.BaseTheme/Assets/Styles/abstracts/mixins";` which both support. (I swear, just when I'm finally rid of Internet Explorer! Is there no escape from Microsoft being the weird one?)

Second, even if you do that, IntelliSense doesn't help with the path! It absolutely refuses to resolve anything outside of the current project directory. You can see in the screenshot that using relative path removes the squiggles from `@include margin-bottom-gutter($multiplier: 1);`, but I have to type in the path manually. Very error prone.
![image](https://user-images.githubusercontent.com/4007293/164891483-8269717a-79c0-4f51-95f0-949406c49b9a.png)

Reply:
Where are we using `.` though?

Please open a VS bug report.

Reply:
Let's keep the original, compact code then, but please add a note about VS and link that bug report here, as well as in our wiki.